web/js/precallController.js

```diff
@@ -100,12 +86,39 @@
           var userNameInputElement = document.getElementById('user-name-input');
           userNameInputElement.value = username;
           userNameInputElement.setAttribute('readonly', true);
+          document.body.addEventListener('keypress', function (event) {
+            enterKeypressHandler(event, resolve);
+          });
         }
       }
       Modal.show(selector, loadModalText);
     });
   }
 
+  var formSubmittedHandler = function (event, resolve) {
+    event.preventDefault();
+    PrecallView.hide();
+    otNetworkTest.stopTest();
+    Modal.hide(selector)
+      .then(function () {
+        publisherOptions.name = document.querySelector(selector
+          + ' input').value.trim();
+        resolve({
+          username: document.querySelector(selector + ' input').value.trim(),
+          publisherOptions: publisherOptions
+        });
+      });
+  };
+
+  var enterKeypressHandler = function (event, resolve) {
+    var keyCode = event.which || event.keyCode;
+    if (keyCode === 13) {
```

Comment:
So instead of catching the key, I think we could focus on the button so hitting enter submits the form, we can make it conditional and focus the username input or the submit button depending on whether there's a username or not.
We might also want to reset the focus after any interaction like toggling mic/video.

Alternatively we can stick with this approach but I'm pretty sure I've read that `keyCode === 13` isn't universal cross-browser

Reply:
If we do decide to go with this I'd also consider writing a robost function to catch the enter key in browser_utils instead of here.